File: notifications-polyglot/java/src/main/java/com/example/notification/service/NotificationService.java
package com.example.notification.service;

import com.example.notification.dto.NotificationRequest;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Service;

import java.util.UUID;

@Service
@RequiredArgsConstructor
@Slf4j
public class NotificationService {

    private final KafkaTemplate<String, Object> kafkaTemplate;
    private final NotificationStatusService statusService;
    private static final String TOPIC = "notifications";

    public String enqueueNotification(NotificationRequest request) {
        String notificationId = UUID.randomUUID().toString();
        log.info("Enqueuing notification: {}", notificationId);
        
        statusService.createInitialStatus(notificationId, request.getChannels());
        
        kafkaTemplate.send(TOPIC, notificationId, request);
        return notificationId;
    }
}

File: notifications-polyglot/kotlin/src/main/kotlin/com/example/notification/service/NotificationService.kt
package com.example.notification.service

import com.example.notification.dto.NotificationRequest
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import org.slf4j.LoggerFactory
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Service
import java.util.UUID

@Service
class NotificationService(
    private val kafkaTemplate: KafkaTemplate<String, Any>,
    private val statusService: NotificationStatusService // Added statusService
) {
    private val logger = LoggerFactory.getLogger(NotificationService::class.java) // Changed logger initialization
    private val topic = "notifications"

    suspend fun enqueueNotification(request: NotificationRequest): String { // Removed withContext(Dispatchers.IO)
        val notificationId = UUID.randomUUID().toString()
        logger.info("Enqueuing notification: {}", notificationId)
        
        // In a real app, we would save to DB here first with status PENDING
        statusService.createInitialStatus(notificationId, request.channels) // Added call to createInitialStatus
        
        kafkaTemplate.send(topic, notificationId, request) // Removed .get()
        return notificationId
    }
}
